scipy/signal/_support_alternative_backends.py

```diff
@@ -0,0 +1,66 @@
+import sys
+import os
+import functools
+from scipy._lib._array_api import array_namespace, is_cupy, is_jax, scipy_namespace_for
+
+from ._signaltools import convolve, convolve2d
+
+
+MODULE_NAME = 'signal'
+_SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", False)
+
+
+def dispatch_xp(dispatcher, module_name):
+    def inner(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwds):
+            xp = dispatcher(*args, **kwds)
+
+            # try delegating to a cupyx/jax namesake
+            if is_cupy(xp):
+                # https://github.com/cupy/cupy/issues/8336
+                import importlib
+                cupyx_module = importlib.import_module(f"cupyx.scipy.{module_name}")
+                cupyx_func = getattr(cupyx_module, func.__name__)
+                return cupyx_func(*args, **kwds)
+            elif is_jax(xp):
+                spx = scipy_namespace_for(xp)
+                jax_module = getattr(spx, module_name)
+                jax_func = getattr(jax_module, func.__name__)
+                return jax_func(*args, **kwds)
+            else:
+                # the original function
+                return func(*args, **kwds)
+        return wrapper
+    return inner
+
+
+
+# X_dispatcher signature must match the signature of X
+
+def convolve_dispatcher(in1, in2, mode='full', method='auto'):
+    xp = array_namespace(in1, in2)
+    return xp
+
+
+def convolve2d_dispatcher(in1, in2, mode='full', boundary='fill', fillvalue=0):
+    xp = array_namespace(in1, in2)
+    return xp
+
+
+# functions we patch for dispatch
+_FUNC_MAP = {
+    convolve: convolve_dispatcher,
+    convolve2d: convolve2d_dispatcher,
+}
+
+
+# ### decorate ###
+for func in _FUNC_MAP:
+    f = (dispatch_cupy(_FUNC_MAP[func], MODULE_NAME)(func)
```

Comment:
I'm seeing many failures/errors locally that look like this:

```
<snip>
scipy/signal/_support_alternative_backends.py:60: in <module>
    f = (dispatch_cupy(_FUNC_MAP[func], MODULE_NAME)(func)
E   NameError: name 'dispatch_cupy' is not defined
<snip>
```

Have you tried this incantation locally: `SCIPY_DEVICE=cuda python dev.py test -j 32 -b all`

Reply:
Sorry, it's just me -- a stray `dispatch_cupy` was left.  Should be fixed now.